File: BinarySearch.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        first = 0
        last = len(nums) - 1
       
        while first <= last:
            pos = (first + last) // 2
            if target == nums[pos]:
                return pos
            elif target > nums[pos]:
                first = pos + 1
            elif target < nums[pos]: 
                last = pos - 1           
        return -1

File: javascript solutions/Binary Search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let start = 0;
    let end = nums.length - 1;
    let mid = 0;
    
    while (start <= end) {
        
        mid = (start + end) // 2;
        
        if (target > nums[mid]){
            start = mid + 1;
        } else if (target < nums[mid]) {
            end = mid - 1;
        } else if (target == nums[mid]) {
            return mid;
        }       
    }
    return -1;
};
